components/style/properties/longhand/box.mako.rs

```diff
@@ -1410,56 +1371,45 @@ ${helpers.predefined_type("scroll-snap-coordinate",
                         if values.len() != 16 {
                             return Err(())
                         }
-                        result.push(SpecifiedOperation::Matrix(
-                            SpecifiedMatrix {
-                                m11: values[ 0], m12: values[ 1], m13: values[ 2], m14: values[ 3],
-                                m21: values[ 4], m22: values[ 5], m23: values[ 6], m24: values[ 7],
-                                m31: values[ 8], m32: values[ 9], m33: values[10], m34: values[11],
-                                m41: values[12], m42: values[13], m43: values[14], m44: values[15]
-                            }));
+                        result.push(SpecifiedOperation::Matrix3D {
+                            m11: values[ 0], m12: values[ 1], m13: values[ 2], m14: values[ 3],
+                            m21: values[ 4], m22: values[ 5], m23: values[ 6], m24: values[ 7],
+                            m31: values[ 8], m32: values[ 9], m33: values[10], m34: values[11],
+                            m41: values[12], m42: values[13], m43: values[14], m44: values[15]
+                            });
                         Ok(())
                     }))
                 },
                 "translate" => {
                     try!(input.parse_nested_block(|input| {
-                        let (tx, ty) = try!(parse_two_lengths_or_percentages(context, input));
-                        result.push(SpecifiedOperation::Translate(TranslateKind::Translate,
-                                                                  tx,
-                                                                  ty,
-                                                                  specified::Length::zero()));
+                        let sx = try!(specified::LengthOrPercentage::parse(context, input));
+                        if let Ok(()) = input.expect_comma() {
```

Comment:
This needs to be `input.try(|input| input.expect_comma())` instead, otherwise you're accepting stuff like `translate(2px foo)`, and parsing it as `translate(2px)`.